pwndbg/heap/ptmalloc.py

```diff
@@ -177,21 +179,61 @@ def bk_nextsize(self):
 
         return self._bk_nextsize
 
+    @property
+    def arena(self):
+        if self._arena is None:
+            try:
+                ar_ptr = pwndbg.heap.current.get_heap(self.address)["ar_ptr"]
+                ar_ptr.fetch_lazy()
+            except Exception:
+                ar_ptr = None
+            if ar_ptr is not None and ar_ptr in (ar.address for ar in pwndbg.heap.current.arenas):
+                self._arena = Arena(ar_ptr)
+            else:
+                self._arena = Arena(pwndbg.heap.current.main_arena.address)
+
+        return self._arena
+
+    @property
+    def is_top_chunk(self):
+        if self._is_top_chunk is None:
+            ar = self.arena
+            if ar is not None and self.address == ar.top:
+                self._is_top_chunk = True
+            else:
+                self._is_top_chunk = False
+
+        return self._is_top_chunk
+
     # TODO Other useful methods e.g. next_chunk(), __iter__, __str__
 
 
 class Arena:
-    def __init__(self, addr, heaps):
-        self.addr = addr
+    def __init__(self, addr, heaps=None):
+        if type(pwndbg.heap.current.malloc_state) == gdb.Type:
+            self._gdbValue = pwndbg.gdblib.memory.poi(pwndbg.heap.current.malloc_state, addr)
+        else:
+            self._gdbValue = pwndbg.heap.current.malloc_state(addr)
+        self.address = int(self._gdbValue.address)
+        self.is_main_arena = self.address == pwndbg.heap.current.main_arena.address
+        self._top = None
         self.heaps = heaps
 
+    @property
+    def top(self):
+        if self._top is None:
+            try:
+                self._top = int(self._gdbValue["top"])
+            except gdb.MemoryError:
+                pass
+
+        return self._top
+
     def __str__(self):
         res = []
         prefix = "[%%%ds]    " % (pwndbg.gdblib.arch.ptrsize * 2)
         prefix_len = len(prefix % (""))
-        arena_name = (
-            hex(self.addr) if self.addr != pwndbg.heap.current.main_arena.address else "main"
-        )
+        arena_name = "main" if self.is_main_arena else hex(self.address)
         res.append(message.hint(prefix % (arena_name)) + str(self.heaps[0]))
```

Comment:
This is picky but if we are here, we could change this to:

```suggestion
        res = [message.hint(prefix % (arena_name)) + str(self.heaps[0])]
```